File: Arrays/905. Sort Array By Parity.py
# url - https://leetcode.com/problems/sort-array-by-parity/

class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        oddPointer = 0
        evenPointer = len(nums) - 1
        
        while oddPointer < evenPointer:
            if nums[oddPointer] % 2 > nums[evenPointer] % 2:
                nums[oddPointer], nums[evenPointer] = nums[evenPointer], nums[oddPointer]
                
            if nums[oddPointer] % 2 == 0:
                oddPointer += 1
            if nums[evenPointer] % 2 == 1:
                evenPointer -= 1  
                            
        return nums       

File: Arrays/905. Sort Array By Parity.cpp
// Url - https://leetcode.com/problems/sort-array-by-parity/

class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int oddPointer = 0;
        int evenPointer = nums.size() - 1;
        
        while(oddPointer < evenPointer)
        {
            while(nums[oddPointer]%2 != 1 && oddPointer < evenPointer)
            {
                oddPointer++;
            }
            while(nums[evenPointer]%2 != 0 && oddPointer < evenPointer)
            {
                evenPointer--;
            }
            int temp = nums[oddPointer];
            nums[oddPointer] = nums[evenPointer];
            nums[evenPointer] = temp;
        }
        
        return nums;
    }
};